src/main/java/seedu/duke/data/TransactionList.java

```diff
@@ -92,12 +99,25 @@ public String addIncome(String description, int amount, String category, LocalDa
         return income.toString();
     }
 
+    public void addIncomeDuringStorage(String description, int amount, String category, LocalDate date) {
+        Income income = new Income(description, amount, category, date);
+        transactions.add(income);
+    }
+
+
+    public void addExpenseDuringStorage(String description, int amount, String category, LocalDate date) {
+        Expense expense = new Expense(description, amount, category, date);
+        transactions.add(expense);
+    }
```

Comment:
Is there really a need for a separate method from addIncome() and addExpense()?

Reply:
Honestly , I dont think theres a need, but I saw that the original addExpense() and addIncome() methods return a String value. Though I could technically use the original methods without dealing with the return value and leave the value unhandled, I created the new methods without a return so its more "correct" since there's no return value to handle in Storage.java, and there would not be any warnings flagged out.